mingpt/model.py

```diff
@@ -189,9 +204,31 @@ def forward(self, idx, targets=None):
         x = self.ln_f(x)
         logits = self.head(x)
 
-        # if we are given some desired targets also calculate the loss
-        loss = None
-        if targets is not None:
-            loss = F.cross_entropy(logits.view(-1, logits.size(-1)), targets.view(-1))
+        return logits
+
+    def step_(self, split, batch, batch_idx=None):
+        idx, targets = batch
+
+        # forward pass the inputs
+        logits = self(idx)
+        # calculate the loss w.r.t. the desired targets
+        loss = F.cross_entropy(logits.view(-1, logits.size(-1)), targets.view(-1))
+
+        if split == 'train':
+            result = pl.TrainResult(minimize=loss)
+        elif split == 'val':
+            result = pl.EvalResult(checkpoint_on=loss)
+        elif split == 'test':
+            result = pl.EvalResult()
+        result.log(f'{split}_loss', loss)
+
+        return result
```

Comment:
if you are not logging, then you can just directly return the loss. no need for a resultsdict